docs/content/doc/advanced/migrations.en-us.md

```diff
@@ -0,0 +1,71 @@
+---
+date: "2019-04-15T17:29:00+08:00"
+title: "Advanced: Migrations Interfaces"
+slug: "migrations-interfaces"
+weight: 30
+toc: true
+draft: false
+menu:
+  sidebar:
+    parent: "advanced"
+    name: "Migrations Interfaces"
+    weight: 55
+    identifier: "migrations-interfaces"
+---
+
+# Migration Features
+
+The new migration features introduced in Gitea 1.9.0. It defined two interfaces to support migrating 
+repositories data from other git host platforms to gitea or in future migrating gitea data to other 
+git host platform. Currently, it only implements to migrate from github via APIv3 to Gitea.
+
+First of all, Gitea defines some standard objects, `Repository`, `Milestone`, `Release`, `Label`, `Issue`,
+`Comment`, `PullRequest`.
+
+## Downloader Interfaces
+
+To migrate from a new git host platform, there are two steps to be updated.
+
+- You should implement a Downloader which could get repository all kinds of informations.
+- You should implement a DownloaderFactory which could detect if the url should match the 
+factory and new a Downloader
+- You should RegisterDownloaderFactory when init
```

Comment:
This makes not much sense. Do you want to say something like `You'll need to register the DownloadFactory via RegisterDownloaderFactory on init`?